src/test.js

```diff
@@ -746,10 +746,15 @@ function makeEachTestName( testName, argument ) {
 function runEach( data, eachFn ) {
 	if ( Array.isArray( data ) ) {
 		data.forEach( eachFn );
+	} else if ( typeof data === "object" && data !== null ) {
+		const keys = Object.keys( data );
+		keys.forEach( ( key ) => {
+			eachFn( data[ key ], key );
+		} );
 	} else {
 		throw new Error(
-			`test.each expects an array of arrays or an array of primitives as the expected input.
-${typeof data} was found instead.`
+			`test.each expects an array of arrays, an array of primitives, or an object 
```

Comment:
I think the array/primitive distinction might be best omitted here now that we don't use variadic args. Something like "... expect an array or object as input."

Reply:
After rebasing with your changes, rewrote this as "Array or object of data values passed to each test case" let me know if that works for you.

Reply:
Sounds good to me! I don't see the change applied, however.